### PR TITLE
Time series Transformers blog post

### PR DESCRIPTION
**Pull request by kashif**
Notebooks are at: 

huggingface/notebooks#272 and huggingface/notebooks#273

### CODE REVIEW

File: time-series-transformers.md [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+
+
+## Introduction
+
+Time series forecasting is an essential scientific and business machine learning problem and as such has also seen a lot of innovation recently with the use of [Deep Learning based global](https://dl.acm.org/doi/abs/10.1145/3533382) or single models in comparison to the [classical methods](https://otexts.com/fpp3/) which are typically classified as local or in other words fitted on each time series in a dataset individually.
+
+When dealing with a large amount of time series for an application, it is typically more efficient to train a single deep learning model so that it can benefit from learned representations from all the time series at inference time in order to make better forecasts. These forecasts are typically point valued and models are trained by minimizing an L2 or L1 type losses with respect to the ground truth data. However, since these forecasts are most probably used in some real-world decision making pipeline, even with humans in the loop, it is much more beneficial to provide the (aleatoric) uncertainties of predictions. Thus *probabilistic* forecasting methods are deployed which hope to learn the distribution of the future values.
```
Comment (NielsRogge) [inline time-series-transformers.md:13]:
```suggestion
When dealing with a large amount of time series for an application, it is typically more efficient to train a single deep learning model so that it can benefit from learned representations from several related time series at inference time in order to make better forecasts. These forecasts are typically point valued and models are trained by minimizing an L2 or L1 type of loss with respect to the ground truth data. However, since these forecasts are most probably used in some real-world decision making pipeline, even with humans in the loop, it is much more beneficial to provide the (aleatoric) uncertainties of predictions. Thus *probabilistic* forecasting methods are deployed which hope to learn the distribution of the future values.
```

File: time-series-transformers.md [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+title: "Time Series Forecasting with 🤗 Transformers"
+thumbnail: /blog/assets/time-series-transformers/thumbnail.png
+---
+
+# Probabilistic Forecasting with Transformers
+
+
+## Introduction
+
+Time series forecasting is an essential scientific and business machine learning problem and as such has also seen a lot of innovation recently with the use of [Deep Learning based global](https://dl.acm.org/doi/abs/10.1145/3533382) or single models in comparison to the [classical methods](https://otexts.com/fpp3/) which are typically classified as local or in other words fitted on each time series in a dataset individually.
```
Comment (NielsRogge) [inline time-series-transformers.md:11]:
```suggestion
Time series forecasting is an essential scientific and business problem and as such has also seen a lot of innovation recently with the use of [Deep Learning based](https://dl.acm.org/doi/abs/10.1145/3533382) models in comparison to the [classical methods](https://otexts.com/fpp3/). Classical methods like ARIMA are typically called "local" or "single" methods as they are fitted on each time series in a dataset individually.
```

File: time-series-transformers.md [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+
+It is common in the probabilistic setting to for example learn the future parameters of some chosen parametric distribution like Gaussian or Student-T, or learn the conditional quantile function or use the framework of Conformal Prediction adapted to the time series setting. The choice of method  does not effect the modeling aspect of methods and thus can be typically thought of as yet another hyperparameter. Empirical means or medians can also provide the point forecasting outputs if desired.
```
Comment (NielsRogge) [inline time-series-transformers.md:15]:
```suggestion
It is common in the probabilistic setting to for example learn the future parameters of some chosen parametric distribution like Gaussian or Student-T, or learn the conditional quantile function or use the framework of Conformal Prediction adapted to the time series setting. The choice of method does not effect the modeling aspect and thus can be typically thought of as yet another hyperparameter. Empirical means or medians can also provide the point forecasting outputs if desired.
```

File: time-series-transformers.md [hunk 4]
```diff
@@ -0,0 +1,789 @@
...
+
+## Transformer for Forecasting
+
+As mentioned, the Encoder-Decoder based Transformer is a natural choice of forecasting as it encapsulates a number of inductive biases nicely. To begin with the use of an Encoder-Decoder is helpful at inference time where typically for some logged data we wish to know the forecast some prediction steps into the future. This can be thought of as analogous to the text generation task where given some context we autoregressively sample the next token. Similarly here we can also, given some distribution type, sample from it autoregressively to return forecasts up till our desired prediction horizon.
```
Comment (NielsRogge) [inline time-series-transformers.md:21]:
```suggestion
As mentioned, the Encoder-Decoder based Transformer is a natural choice of forecasting as it encapsulates a number of inductive biases nicely. To begin with, the use of an Encoder-Decoder architecture is helpful at inference time where typically for some logged data we wish to know the forecast some prediction steps into the future. This can be thought of as analogous to the text generation task where given some context we autoregressively sample the next token. Similarly here we can also, given some distribution type, sample from it autoregressively to return forecasts up until our desired prediction horizon.
```

File: time-series-transformers.md [hunk 5]
```diff
@@ -0,0 +1,789 @@
...
+
+Secondly, this paradigm helps us to train on time series data which might contain thousands of time points. It might not be feasible to input *all* the history of a time series  at once to the model and thus one can consider some appropriate context window and sample this window and the subsequent prediction length sized window from the training data when constructing batches for SGD. The context sized window can be passed to the encoder and the prediction window to a *masked* decoder. 
+
+Another benefit of Transformers over the other architectures is that we can incorporate missing values (which are common in the time series setting) as additional mask to the encoder or decoder and still train without resorting to in-filling or imputation.
```
Comment (NielsRogge) [inline time-series-transformers.md:25]:
```suggestion
Another benefit of Transformers over the other architectures is that we can incorporate missing values (which are common in the time series setting) as an additional mask to the encoder or decoder and still train without resorting to in-filling or imputation. This is equivalent to the `attention_mask` of models like BERT and GPT-2 in the Transformers library, to not include padding tokens in the computation of the attention matrix.
```

File: time-series-transformers.md [hunk 6]
```diff
@@ -0,0 +1,789 @@
...
+
+A drawback of this architecture is that obviously there is a limit to the sizes of the context and prediction windows in view of the quadratic compute and memory requirements of the vanilla Transformer and also since the Transformer is a power architecture it might overfit or learn spurious correlations much more easily compared to other methods.
+
+The 🤗 Transformers library comes with a vanilla probabilistic time series Transformer model which is able to learn a number of real and integer valued parametric distributions together with all the features of neural forecasting which we will highlight below!
+
+## Set-up environment
+
+First, let's install the necessary libraries: 🤗 Transformers, 🤗 Datasets, 🤗 Evaluate,  🤗 Accelerate and Gluon-ts.
```
Comment (NielsRogge) [inline time-series-transformers.md:33]:
```suggestion
First, let's install the necessary libraries: 🤗 Transformers, 🤗 Datasets, 🤗 Evaluate, 🤗 Accelerate and Gluon-ts. We'll use the latter for preparing the data for the model.
```

File: time-series-transformers.md [hunk 7]
```diff
@@ -40,15 +40,15 @@ Typically, classical methods are fitted on each time series in a dataset individ
...
 
 ## The Time Series Transformer
 
 In terms of modeling time series data which are sequential in nature, as one can imagine, researchers have come up with models which use Recurrent Neural Networks (RNN) like LSTM or GRU, or Convolutional Networks (CNN), and more recently Transformer based methods which fit naturally to the time series forecasting setting.
 
-In this blog post, we're going to leverage the vanilla Transformer [(Vaswani et al., 2017)](https://arxiv.org/abs/1706.03762) for  probabilistic forecasting. The Encoder-Decoder Transformer is a natural choice for forecasting as it encapsulates several inductive biases nicely. 
+In this blog post, we're going to leverage the vanilla Transformer [(Vaswani et al., 2017)](https://arxiv.org/abs/1706.03762) for  the **univariate** probabilistic forecasting task. The Encoder-Decoder Transformer is a natural choice for forecasting as it encapsulates several inductive biases nicely.
```
Comment (NielsRogge) [inline time-series-transformers.md:51]:
```suggestion
In this blog post, we're going to leverage the vanilla Transformer [(Vaswani et al., 2017)](https://arxiv.org/abs/1706.03762) for the **univariate** probabilistic forecasting task (in a future blog post, we'll showcase the multivariate case). The Encoder-Decoder Transformer is a natural choice for forecasting as it encapsulates several inductive biases nicely. 
```

File: time-series-transformers.md [hunk 8]
```diff
@@ -40,15 +40,15 @@ Typically, classical methods are fitted on each time series in a dataset individ
 
 Some classical methods are point-valued (meaning, they just output a single value per time step) and models are trained by minimizing an L2 or L1 type of loss with respect to the ground truth data. However, since forecasts are often used in some real-world decision making pipeline, even with humans in the loop, it is much more beneficial to provide the uncertainties of predictions. This is also called "probabilistic forecasting", as opposed to "point forecasting". This entails modeling a probabilistic distribution, from which one can sample.
 
-So in short, rather than training local point forecasting models, we hope to train **probabilistic** models. Deep learning is a great fit for this, as neural networks can learn representations from several related time series, and model the parameters of a distribution.
+So in short, rather than training local point forecasting models, we hope to train **probabilistic** models. Deep learning is a great fit for this, as neural networks can learn representations from several related time series and model the uncertainty of the data (Aleatoric).
```
Comment (NielsRogge) [inline time-series-transformers.md:43]: Yeah I left away the word Aleatoric as I assume many people don't know this word (including me haha)

File: time-series-transformers.md [hunk 9]
```diff
@@ -845,10 +845,12 @@ How do we compare against other models? The [Monash Time Series Repository](http
...
 
-Of course, we need to be careful with just claiming state-of-the-art results on time series with neural networks, as it seems "XGBoost is typically all you need". We are just very curious to see how far neural networks can bring us, and whether Transformers are going to be useful in this domain. This particular dataset seems to indicate that it's definitely worth exploring.
+Of course, we need to be careful with just claiming state-of-the-art results on time series with neural networks, as it seems ["XGBoost is typically all you need"](https://www.sciencedirect.com/science/article/pii/S0169207021001679). More specificity [LightGBM](https://github.com/microsoft/LightGBM), since it can also fit linear models in the leaf-nodes, which makes it more suitable for point forecasting or [LightGBMLSS](https://github.com/StatMixedML/LightGBMLSS) for the probabilistic case, but we digress. We are just very curious to see how far neural networks can bring us, and whether Transformers are going to be useful in this domain. This particular dataset seems to indicate that it's definitely worth exploring.
```
Comment (NielsRogge) [inline time-series-transformers.md:848]: specificity => specifically

File: time-series-transformers.md [hunk 10]
```diff
@@ -845,10 +845,12 @@ How do we compare against other models? The [Monash Time Series Repository](http
...
 
 ## Next Steps
 
+We would encourage the readers to try out the notebook with other time series datasets from the 🤗 [hub](https://huggingface.co/datasets/monash_tsf) and replacing the appropriate frequency and prediction length parameters. For your own datasets one would need to convert it to the convention used by GluonTS which is explained nicely in their documentation [here](https://ts.gluon.ai/stable/tutorials/forecasting/extended_tutorial.html#What-is-in-a-dataset?).
```
Comment (NielsRogge) [inline time-series-transformers.md]: Would it be possible to create a separate notebook on how to, starting from a csv/excel file, turn it into a HF Dataset and push it to the hub?

Comment (kashif) [reply time-series-transformers.md]: yes i can do that

File: time-series-transformers.md [hunk 11]
```diff
@@ -0,0 +1,872 @@
...
+
+Some classical methods are point-valued (meaning, they just output a single value per time step) and models are trained by minimizing an L2 or L1 type of loss with respect to the ground truth data. However, since forecasts are often used in some real-world decision making pipeline, even with humans in the loop, it is much more beneficial to provide the uncertainties of predictions. This is also called "probabilistic forecasting", as opposed to "point forecasting". This entails modeling a probabilistic distribution, from which one can sample.
+
+So in short, rather than training local point forecasting models, we hope to train **probabilistic** models. Deep learning is a great fit for this, as neural networks can learn representations from several related time series and model the uncertainty of the data.
```
Comment (NielsRogge) [inline time-series-transformers.md:48]: So you've left "multivariate" out here? I'm not sure we should talk about the fact that classical methods are "local" models in case our model is also local

File: time-series-transformers.md [hunk 12]
```diff
@@ -0,0 +1,872 @@
...
+
+## The Time Series Transformer
+
+In terms of modeling time series data which are sequential in nature, as one can imagine, researchers have come up with models which use Recurrent Neural Networks (RNN) like LSTM or GRU, or Convolutional Networks (CNN), and more recently Transformer based methods which fit naturally to the time series forecasting setting.
+
+In this blog post, we're going to leverage the vanilla Transformer [(Vaswani et al., 2017)](https://arxiv.org/abs/1706.03762) for the **univariate** probabilistic forecasting task (in a future blog post, we'll showcase the multivariate case). The Encoder-Decoder Transformer is a natural choice for forecasting as it encapsulates several inductive biases nicely.
```
Comment (NielsRogge) [inline time-series-transformers.md:56]: This isn't great as this is the first time we talk about univariate/multivariate. 

People will wonder what these mean, because they are not introduced in the previous section anymore

File: time-series-transformers.md (hunk 11 above)
Comment (NielsRogge) [inline time-series-transformers.md:48]:
```suggestion
So in short, rather than training local point forecasting models, we hope to train **global probabilistic** models. Deep learning is a great fit for this, as neural networks can learn representations from several related time series and model the uncertainty of the data.
```
It'd add "global" here, to make the distinction with the classical local methods

File: time-series-transformers.md (hunk 12 above)
Comment (NielsRogge) [inline time-series-transformers.md:56]: People might wonder here what univariate/multivariate mean

File: time-series-transformers.md [hunk 13]
```diff
@@ -0,0 +1,872 @@
...
+
+Another benefit of Transformers over the other architectures is that we can incorporate missing values (which are common in the time series setting) as an additional mask to the encoder or decoder and still train without resorting to in-filling or imputation. This is equivalent to the `attention_mask` of models like BERT and GPT-2 in the Transformers library, to not include padding tokens in the computation of the attention matrix.
+
+A drawback of the Transformer architecture is the limit to the sizes of the context and prediction windows because of the quadratic compute and memory requirements of the vanilla Transformer. Additionally, since the Transformer is a powerful  architecture it might overfit or learn spurious correlations much more easily compared to other [methods](https://openreview.net/pdf?id=D7YBmfX_VQy).
```
Comment (NielsRogge) [inline time-series-transformers.md:64]:
```suggestion
A drawback of the Transformer architecture is the limit to the sizes of the context and prediction windows because of the quadratic compute and memory requirements of the vanilla Transformer. Additionally, since the Transformer is a powerful architecture, it might overfit or learn spurious correlations much more easily compared to other [methods](https://openreview.net/pdf?id=D7YBmfX_VQy).
```

Comment (NielsRogge) [inline time-series-transformers.md:64]:
```suggestion
A drawback of the Transformer architecture is the limit to the sizes of the context and prediction windows because of the quadratic compute and memory requirements of the vanilla Transformer, see [Tay et al., 2020](https://arxiv.org/abs/2009.06732). Additionally, since the Transformer is a powerful  architecture it might overfit or learn spurious correlations much more easily compared to other [methods](https://openreview.net/pdf?id=D7YBmfX_VQy).
```

File: time-series-transformers.md [hunk 14]
```diff
@@ -0,0 +1,872 @@
...
+
+
+Each example contains a few keys, of which `start` and `target` are the most important ones. Let us have a look at the first time series in the dataset:
+
+
+```python
+train_example = dataset['train'][0]
+train_example.keys()
+
+    dict_keys(['start', 'target', 'feat_static_cat', 'feat_dynamic_real', 'item_id'])
```
Comment (NielsRogge) [inline time-series-transformers.md:132]: Not sure this will render properly

File: time-series-transformers.md [hunk 15]
```diff
@@ -0,0 +1,872 @@
...
+The `start` simply indicates the start of the time series (as a datetime), and the `target` contains the actual values of the time series.
+
+The `start` will be useful to add time related features to the time series values, as extra input to the model (such as "month of year"). Since we know the frequency of the data is `monthly`, we know for instance that the second value has the timestamp `1979-02-01`, etc.
+
+
+```python
+print(train_example['start'])
+print(train_example['target'])
+
+    1979-01-01 00:00:00
```
Comment (NielsRogge) [inline time-series-transformers.md:146]: Same here, can you look into how other blogs render the result of a code statement? See e.g. this one: https://huggingface.co/blog/how-to-generate

File: time-series-transformers.md [hunk 16]
```diff
@@ -0,0 +1,872 @@
...
+    [1149.8699951171875, 1053.8001708984375, ..., 5772.876953125]
+```
+
+
+The validation set contains the same data as the training set, just for a `prediction_length` longer amount of time. This allows us to validate the model's predictions against the ground truth.
+
+The test set is again one `prediction_length` longer data compared to the validation set (or some multiple of  `prediction_length` longer data compared to the training set for testing on multiple rolling windows).
```
Comment (NielsRogge) [inline time-series-transformers.md:153]:
```suggestion
The test set is again one `prediction_length` longer data compared to the validation set (or some multiple of `prediction_length` longer data compared to the training set for testing on multiple rolling windows).
```

File: time-series-transformers.md [hunk 17]
```diff
@@ -0,0 +1,872 @@
...
+```python
+from functools import partial
+
+train_dataset.set_transform(partial(transform_start_field, freq=freq, log1p=True))
+val_dataset.set_transform(partial(transform_start_field, freq=freq, log1p=True))
+```
+
+## Define the Model
+
+Next, let's instantiate a model. The model will be trained from scratch, hence we won't use the `from_pretrained` method here, but rather instantiate the model from a [`config`](https://huggingface.co/docs/transformers/model_doc/time_series_transformer#transformers.TimeSeriesTransformerConfig).
```
Comment (NielsRogge) [inline time-series-transformers.md:244]:
```suggestion
Next, let's instantiate a model. The model will be trained from scratch, hence we won't use the `from_pretrained` method here, but rather randomly initialize the model from a [`config`](https://huggingface.co/docs/transformers/model_doc/time_series_transformer#transformers.TimeSeriesTransformerConfig).
```

File: time-series-transformers.md [hunk 18]
```diff
@@ -0,0 +1,872 @@
...
+
+We specify a couple of additional parameters to the model:
+- `prediction_length` (in our case, `24` months): this is horizon the decoder of the Transformer will learn to predict for;
```
Comment (NielsRogge) [inline time-series-transformers.md:247]:
```suggestion
- `prediction_length` (in our case, `24` months): this is the horizon that the decoder of the Transformer will learn to predict for;
```

File: time-series-transformers.md [hunk 19]
```diff
@@ -0,0 +1,872 @@
...
+- `context_length`: the model will set the `context_length` (input of the encoder) equal to this, if no `context_length` is specified;
+- `lags` for a given frequency: these specify how much we "look back", to be added as additional features. e.g. for a `Daily` frequency we might consider a look back of `[1, 2, 7, 30, ...]` or in other words look back 1, 2, ... days) while for `Minute` data we might consider `[1, 30, 60, 60*24, ...]` etc.;
+- the number of time features: in our case, this will be `2` as we'll add `MonthOfYear` as well as an `Age` feature;
+- the number of static categorical features: in our case, this will be just `1` as we'll add a single "time series ID" feature;
+- the cardinality: the number of values of each categorical feature, as a list which for our case will be `[366]` as we have 366 different time series
+- the embedding dimension: the embedding dimension for each categorical feature, as a list, for example `[2]` meaning the model will learn an embedding for vector of size `2` for each of the `366` time series.
```
Comment (NielsRogge) [inline time-series-transformers.md:253]:
```suggestion
- the embedding dimension: the embedding dimension for each categorical feature, as a list, for example `[2]` meaning the model will learn an embedding vector of size `2` for each of the `366` time series (regions).
```

File: time-series-transformers.md [hunk 20]
```diff
@@ -0,0 +1,872 @@
...
+
+Note that, similar to other models in the 🤗 Transformers library, [`TimeSeriesTransformerModel`](https://huggingface.co/docs/transformers/model_doc/time_series_transformer#transformers.TimeSeriesTransformerModel) corresponds to the encoder-decoder Transformer without any head on top, and [`TimeSeriesTransformerForPrediction`](https://huggingface.co/docs/transformers/model_doc/time_series_transformer#transformers.TimeSeriesTransformerForPrediction) corresponds to `TimeSeriesTransformerModel` with a **distribution head** on top. By default, the model uses a Student-t distribution (but this is configurable):
+
+```python
+model.config.distribution_output
+
+    student_t
+```
+
+This is an important difference with Transformers for NLP, where the head typically consists of a fixed categorical distribution.
```
Comment (NielsRogge) [inline time-series-transformers.md:313]:
```suggestion
This is an important difference with Transformers for NLP, where the head typically consists of a fixed categorical distribution implemented as an `nn.Linear` layer.
```

File: time-series-transformers.md [hunk 21]
```diff
@@ -0,0 +1,872 @@
...
+        "train": train_sampler or ExpectedNumInstanceSampler(
+            num_instances=1.0, min_future=config.prediction_length
+        ),
+        "validation":  validation_sampler or ValidationSplitSampler(
+            min_future=config.prediction_length
+        ),
+        "test": TestSplitSampler(),
+    }[mode]
+
+    return InstanceSplitter(
```
Comment (NielsRogge) [inline time-series-transformers.md:462]: This doesn't seem right to me, we're importing InstanceSampler from gluon but here we're defining an InstanceSplitter

Can we use the same term (sampler/splitter) everywhere?

File: time-series-transformers.md [hunk 22]
```diff
@@ -0,0 +1,872 @@
...
+|Tourism Monthly | 	3.306 |	1.649 |	1.751 |	1.526|	1.589|	1.678	|1.699|	1.582	| 1.409	| 1.574|	1.482	|  **0.863**|
+
+Note that, with our model, we are beating all other models reported at https://forecastingdata.org/#results (see also table 2 in the corresponding [paper](https://openreview.net/pdf?id=wEc1mgAjU-)) by quite a margin, and we didn't do any hyperparameter tuning. We just trained the Transformer for 60 epochs. 
+
+Of course, we need to be careful with just claiming state-of-the-art results on time series with neural networks, as it seems ["XGBoost is typically all you need"](https://www.sciencedirect.com/science/article/pii/S0169207021001679). More specifically [LightGBM](https://github.com/microsoft/LightGBM), since it can also fit linear models in the leaf-nodes, which makes it more suitable for point forecasting or [LightGBMLSS](https://github.com/StatMixedML/LightGBMLSS) for the probabilistic case, but we digress. We are just very curious to see how far neural networks can bring us, and whether Transformers are going to be useful in this domain. This particular dataset seems to indicate that it's definitely worth exploring.
```
Comment (NielsRogge) [inline time-series-transformers.md:857]:
```suggestion
Of course, we need to be careful with just claiming state-of-the-art results on time series with neural networks, as it seems ["XGBoost is typically all you need"](https://www.sciencedirect.com/science/article/pii/S0169207021001679). We are just very curious to see how far neural networks can bring us, and whether Transformers are going to be useful in this domain. This particular dataset seems to indicate that it's definitely worth exploring.
```
Wouldn't make this comment too long

File: time-series-transformers.md [hunk 23]
```diff
@@ -0,0 +1,872 @@
...
+
+## Next Steps
+
+We would encourage the readers to try out the notebook with other time series datasets from the 🤗 [hub](https://huggingface.co/datasets/monash_tsf) and replace the appropriate frequency and prediction length parameters. For your own datasets one would need to convert it to the convention used by GluonTS which is explained nicely in their documentation [here](https://ts.gluon.ai/stable/tutorials/forecasting/extended_tutorial.html#What-is-in-a-dataset?). We have also prepared an example notebook showing you how to convert your dataset into the 🤗 datasets format [here](https://github.com/huggingface/notebooks/blob/main/examples/time_series_datasets.ipynb).
```
Comment (NielsRogge) [inline time-series-transformers.md:861]:
```suggestion
We would encourage the readers to try out the [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/time-series-transformers.ipynb) with other time series datasets from the 🤗 [hub](https://huggingface.co/datasets/monash_tsf) and replace the appropriate frequency and prediction length parameters. For your own datasets one would need to convert it to the convention used by GluonTS which is explained nicely in their documentation [here](https://ts.gluon.ai/stable/tutorials/forecasting/extended_tutorial.html#What-is-in-a-dataset?). We have also prepared an example notebook showing you how to convert your dataset into the 🤗 datasets format [here](https://github.com/huggingface/notebooks/blob/main/examples/time_series_datasets.ipynb).
```

File: time-series-transformers.md (hunk 21 above)
Comment (kashif) [reply time-series-transformers.md:462]: we are importing both the InstanceSplitter and InstanceSampler from gluonts and the InstanceSplitter is the main transformation and it takes an InstanceSampler as input... but yes we are importing both and only talk about the InstanceSampler in the text

Comment (NielsRogge) [reply time-series-transformers.md:462]: I see!

File: time-series-transformers.md [hunk 24]
```diff
@@ -0,0 +1,872 @@
+---
+title: "Probabilistic Time Series Forecasting with 🤗 Transformers"
+thumbnail: /blog/assets/118_time-series-transformers/thumbnail.png
```
Comment (osanseviero) [inline time-series-transformers.md:3]: The thumbnail might be a bit hard to read

File: time-series-transformers.md [hunk 25]
```diff
@@ -0,0 +1,872 @@
...
+---
```
Comment (osanseviero) [inline time-series-transformers.md:4]: For non-thmbnail assets, feel free to have them in https://huggingface.co/datasets/huggingface/documentation-images

File: time-series-transformers.md [hunk 26]
```diff
@@ -0,0 +1,872 @@
...
+
+<h1>Probabilistic Time Series Forecasting with 🤗 Transformers</h1>
+
+<div class="blog-metadata">
+    <small>Published November 23, 2022.</small>
```
Comment (osanseviero) [inline time-series-transformers.md:9]: To update

File: time-series-transformers.md [hunk 27]
```diff
@@ -0,0 +1,872 @@
...
+
+<script async defer src="https://unpkg.com/medium-zoom-element@0/dist/medium-zoom-element.min.js"></script>
+
+<a target="_blank" href="https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/time-series-transformers.ipynb">
+    <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
+</a>
+
+## Introduction
+
+Time series forecasting is an essential scientific and business problem and as such has also seen a lot of innovation recently with the use of [Deep Learning based](https://dl.acm.org/doi/abs/10.1145/3533382) models in addition to the [classical methods](https://otexts.com/fpp3/). An important difference between classical methods like ARIMA and novel deep learning methods is the following.
```
Comment (osanseviero) [inline time-series-transformers.md:40]:
```suggestion
Time series forecasting is an essential scientific and business problem and as such has also seen a lot of innovation recently with the use of [deep learning based](https://dl.acm.org/doi/abs/10.1145/3533382) models in addition to the [classical methods](https://otexts.com/fpp3/). An important difference between classical methods like ARIMA and novel deep learning methods is the following.
```

File: time-series-transformers.md [hunk 28]
```diff
@@ -0,0 +1,872 @@
...
+!pip install -q evaluate
+
+!pip install -q accelerate
+
+!pip install -q gluonts ujson
+```
+
+## Load Dataset
+
+In this blog post, we'll use the `tourism_monthly` dataset, which is available on the 🤗 [hub](https://huggingface.co/datasets/monash_tsf). This dataset contains monthly tourism volumes for 366 regions in Australia.
```
Comment (osanseviero) [inline time-series-transformers.md:90]:
```suggestion
In this blog post, we'll use the `tourism_monthly` dataset, which is available on the [Hugging Face Hub](https://huggingface.co/datasets/monash_tsf). This dataset contains monthly tourism volumes for 366 regions in Australia.
```

File: time-series-transformers.md [hunk 29]
```diff
@@ -0,0 +1,867 @@
...
+
+## Next Steps
+
+We would encourage the readers to try out the [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/time-series-transformers.ipynb) with other time series datasets from the 🤗 [hub](https://huggingface.co/datasets/monash_tsf) and replace the appropriate frequency and prediction length parameters. For your own datasets one would need to convert it to the convention used by GluonTS which is explained nicely in their documentation [here](https://ts.gluon.ai/stable/tutorials/forecasting/extended_tutorial.html#What-is-in-a-dataset?). We have also prepared an example notebook showing you how to convert your dataset into the 🤗 datasets format [here](https://github.com/huggingface/notebooks/blob/main/examples/time_series_datasets.ipynb).
```
Comment (osanseviero) [inline time-series-transformers.md:856]:
```suggestion
We would encourage the readers to try out the [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/time-series-transformers.ipynb) with other time series datasets from the [Hub](https://huggingface.co/datasets/monash_tsf) and replace the appropriate frequency and prediction length parameters. For your datasets, one would need to convert them to the convention used by GluonTS, which is explained nicely in their documentation [here](https://ts.gluon.ai/stable/tutorials/forecasting/extended_tutorial.html#What-is-in-a-dataset?). We have also prepared an example notebook showing you how to convert your dataset into the 🤗 datasets format [here](https://github.com/huggingface/notebooks/blob/main/examples/time_series_datasets.ipynb).
```

File: time-series-transformers.md (hunk 25 above)
Comment (kashif) [reply time-series-transformers.md:4]: done
